osquery/extensions/impl_thrift.cpp

```diff
@@ -45,6 +49,46 @@ using TPlatformServerSocket = TServerSocket;
 using TPlatformSocket = TSocket;
 #endif
 
+class ThriftServerEventHandler : public TServerEventHandler,
+                                 boost::noncopyable {
+ public:
+  ThriftServerEventHandler() : server_is_listening_{false} {}
+
+  void preServe() override {
+    std::unique_lock<std::mutex> lock(m_);
```

Comment:
Is the lock only protecting `server_is_listening_`? You can use an `std::atomic<bool>` is that is the case.

Reply:
The lock (and the underlying mutex) is needed to protect concurrent access to the condition variable; it is passed to it in the wait_for:
https://github.com/osquery/osquery/pull/5937/files#diff-5a22bf9f74d2442551babd3d546477cfR70-R72